File: src/commands/cat.ts
import Command from '../command';
import { Trace } from '@aiteq/trace';
import prompts from 'prompts';

@Trace()
export default class Cat extends Command {
  constructor() {
    super('cat');
  }

  private async determineFile() {
    let file: string;

    do {
      const response = await prompts(
        {
          type: 'text',
          name: 'choice',
          message: 'Which file would you like to cat?',
        },
        {
          onCancel: () => process.exit(),
        },
      );

      file = response.choice;
    } while (!file);

    return file;
  }

  public async run() {
    let file;
    const choices = this.generateChoices();
    const choice = await this.ask(choices);
    if (!choice) {
      return;
    }

    this.id = choice.split('-')[0].trim();

    if (this.args.nonFlags.length > 1) {
      // Remove the first element since this is used for fuzzy searching
      this.args.nonFlags.shift();
      file = this.args.nonFlags.join('');
    } else {
      file = await this.determineFile();
    }

    return this.spawn('docker', ['exec', '-ti', this.id.trim(), 'cat', file]);
  }
}

File: src/commands/cat.rs
use crate::utils::{exec, prompt};

/// Ease of use shortcut to view the contents of a file within a running container
pub fn run() {
    let mut args = vec!["exec".to_string(), "-ti".to_string()];

    let choice = prompt::prompt("Select a container from the list", "cat");
    let file = prompt::text("Which file would you like to cat?");
    args.extend(vec![choice, "cat".to_string(), file]);

    exec::run_and_exit(&args);
}
